hudi-client/hudi-client-common/src/test/java/org/apache/hudi/client/transaction/TestTransactionManager.java

```diff
@@ -152,6 +165,32 @@ public void testMultiWriterTransactions() {
     Assertions.assertTrue(writer2Completed.get());
   }
 
+  @Test
+  public void testEndTransactionByDiffOwner() throws InterruptedException {
+    // 1. Begin and end by the same transaction owner
+    Option<HoodieInstant> lastCompletedInstant = getInstant("0000001");
+    Option<HoodieInstant> newTxnOwnerInstant = getInstant("0000002");
+    transactionManager.beginTransaction(newTxnOwnerInstant, lastCompletedInstant);
+
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+    // Another writer thread
+    Thread writer2 = new Thread(() -> {
+      Option<HoodieInstant> newTxnOwnerInstant1 = getInstant("0000003");
+      transactionManager.endTransaction(newTxnOwnerInstant1);
+      countDownLatch.countDown();
+    });
+
+    writer2.start();
+    countDownLatch.await(30, TimeUnit.SECONDS);
+    // should not have reset the state within transaction manager since the owner is different.
+    Assertions.assertTrue(transactionManager.getCurrentTransactionOwner().isPresent());
```

Comment:
iiuc, even in this scenario, without this patch, `reset` would be called and it would return w/o resetting but then `unlock` would also be called which we don't want. And this patch fixes this behavior. So, can we verify that not only state is not reset but also `unlock` is not called?

Reply:
maybe make reset method just visible for testing?

Reply:
we usually don't use VisibleForTesting annotation. lets sync up on this. we might have tests in InProcessLockProvider to assert for the condition you are talking about

Reply:
Ok found it: `TestInProcessLockProvider#testLockReAcquisitionByDifferentThread`. That's good enough. 
Since we already use Mocito, maybe we can use its verify API to just verify calls.